server/api/ballot_manifest.py

```diff
@@ -100,8 +104,23 @@ def process() -> None:
             AuditType.BALLOT_COMPARISON,
             AuditType.HYBRID,
         ]
+
+        is_container_required = is_enabled_sample_extra_batches_by_counting_group(
+            jurisdiction.election
+        )
+        container_values_allowlist = (
+            set(item.value for item in ContainerType)
+            if is_container_required
+            else set()
+        )
+
         columns = [
-            CSVColumnType(CONTAINER, CSVValueType.TEXT, required_column=False),
+            CSVColumnType(
+                CONTAINER,
+                CSVValueType.TEXT,
+                required_column=is_container_required,
+                allowed_values=container_values_allowlist,
+            ),
```

Comment:
While I like the idea of adding this logic to CSV validation, since it's more general and reusable, the resulting error message is not the most helpful. I wonder if we should implement this instead by checking the container value on the parsed rows and returning an error message that's more specific to this situation, e.g. 

> Invalid value for row 4, column Container: [some invalid value]. Use the Batch Audit File Preparation Tool to create your ballot manifest, or correct this value to one of the following: Election Day, Advance Voting [...]

Reply:
Moved into https://github.com/votingworks/arlo/pull/2011/commits/52966ae735c0b44e18058cef22a1e33088a3e3d8#diff-faa72bf6794277d58d8a9a0f53d986cd4efd48c091f5eff1b4c663ff199aa90fR153

This seems safe to do inline because we don't commit `db_session` until later, so encountering a validation error partway through iteration won't result in hanging db objects.